plugins/modules/ndo_mac_sec_policy.py

```diff
@@ -0,0 +1,424 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024, Anvitha Jain (@anvjain) <anvjain@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_mac_sec_policy
+short_description: Manage MACSec Policies on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage MACSec Policies on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Anvitha Jain (@anvjain)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a fabric policy template.
+    type: str
+    required: true
+  mac_sec_policy:
+    description:
+    - The name of the MACSec Policy.
+    type: str
+    aliases: [ name ]
+  mac_sec_policy_uuid:
+    description:
+    - The uuid of the MACSec Policy.
+    - This parameter is required when the O(mac_sec_policy) needs to be updated.
+    type: str
+    aliases: [ uuid ]
+  description:
+    description:
+    - The description of the MACSec Policy.
+    type: str
+  admin_state:
+    description:
+    - The administrative state of the MACSec Policy. (Enables or disables the policy)
+    - The default value is C(enabled).
+    type: str
+    choices: [ enabled, disabled ]
+  interface_type:
+    description:
+    - The type of the interfaces this policy will be applied to.
+    type: str
+    choices: [ fabric, access ]
+    default: fabric
+  cipher_suite:
+    description:
+    - The cipher suite to be used for encryption.
+    - The default value is C(256_gcm_aes_xpn).
+    type: str
+    choices: [ 128_gcm_aes, 128_gcm_aes_xpn, 256_gcm_aes, 256_gcm_aes_xpn ]
+  window_size:
+    description:
+    - The window size defines the maximum number of frames that can be received out of order
+    - before a replay attack is detected.
+    - The value must be between 0 and 4294967295.
```

Comment:
Can you add code that fails the module if window_size is not between 0 and 4294967295?
```
if window_size is not None and window_size not in range(0, 4294967296):
     mso.fail_json(msg="Invalid value provided for window_size: {0}; it must be in the range 0 - 4294967295".format(window_size))
     
```

Reply:
API will error out, if value is out-of-range. Hence, will not be adding additional check as discussed.